tests/Repositories/ReportRepositoryTest.php

```diff
@@ -0,0 +1,188 @@
+    /** @test */
+    public function it_can_retrieve_tag_ids_of_report()
+    {
+        $report = factory(Report::class)->create();
+        $tags = factory(Tag::class)->times(2)->create();
+
+        factory(ReportFilter::class)->create([
+            'report_id'  => $report->id,
+            'param_id'   => $tags[0]->id,
+            'param_type' => Tag::class,
+        ]);
+        factory(ReportFilter::class)->create([
+            'report_id'  => $report->id,
+            'param_id'   => $tags[1]->id,
+            'param_type' => Tag::class,
+        ]);
+
+        $tagIds = $this->reportRepo->getTagIds($report->id);
+
+        $this->assertCount(2, $tagIds);
+
+        $this->assertEquals($tags[0]->id, $tagIds[0]);
+        $this->assertEquals($tags[1]->id, $tagIds[1]);
+    }
+
+    /** @test */
+    public function it_can_retrieve_user_ids_of_report()
+    {
+        $report = factory(Report::class)->create();
+        $users = factory(User::class)->times(2)->create();
+
+        factory(ReportFilter::class)->create([
+            'report_id'  => $report->id,
+            'param_id'   => $users[0]->id,
+            'param_type' => User::class,
+        ]);
+        factory(ReportFilter::class)->create([
+            'report_id'  => $report->id,
+            'param_id'   => $users[1]->id,
+            'param_type' => User::class,
+        ]);
+
+        $userIds = $this->reportRepo->getUserIds($report->id);
+
+        $this->assertCount(2, $userIds);
+
+        $this->assertEquals($users[0]->id, $userIds[0]);
+        $this->assertEquals($users[1]->id, $userIds[1]);
+    }
+
+    /** @test */
+    public function it_can_retrieve_client_id_of_report()
+    {
+        $report = factory(Report::class)->create();
+        $client = factory(Client::class)->create();
+
+        factory(ReportFilter::class)->create([
+            'report_id'  => $report->id,
+            'param_id'   => $client->id,
+            'param_type' => Client::class,
+        ]);
+
+        $clientId = $this->reportRepo->getClientId($report->id);
+
+        $this->assertNotEmpty($clientId);
+        $this->assertEquals($client->id, $clientId);
+    }
+
+    /** @test */
+    public function it_will_return_empty_without_param_type()
+    {
+        $report = factory(Report::class)->create();
+        $client = factory(Client::class)->create();
+
+        factory(ReportFilter::class)->create([
+            'report_id' => $report->id,
+            'param_id'  => $client->id,
+        ]);
+
+        $clientId = $this->reportRepo->getClientId($report->id);
+
+        $this->assertEmpty($clientId);
+    }
+
+    /** @test */
+    public function it_can_delete_report_filter_for_report()
+    {
+        $report = factory(Report::class)->create();
+
+        $reportFilter = factory(ReportFilter::class)->create(['report_id' => $report->id]);
+
+        $deleteFilter = $this->reportRepo->deleteFilter($report->id);
+
+        $this->assertEquals($reportFilter->id, $deleteFilter);
```

Comment:
**THIS IS TOTALLY WRONG** $deleteFilter contain `true` means 1, and report filter id is `1`, that why this test is passing, fix it